### ISSUE TITLE
timeout exception not captured

### ISSUE THREAD

**Issue by javiplx** [javiplx/ameba-C3#6]
```
 Traceback (most recent call last):
  File "/usr/bin/aupd", line 464, in <module>
    if not command[ subcmd ]( config , options , args ) :
  File "/usr/bin/aupd", line 337, in pull
    ret = amebaC3_client.client.pull ( _url , _uuid , _cmds.split(',') , outdated_retcode )
  File "/usr/lib/python2.5/site-packages/amebaC3_client/client.py", line 125, in pull
    sessid , delay = login( url , uuid )
  File "/usr/lib/python2.5/site-packages/amebaC3_client/client.py", line 64, in login
    res = urllib2.urlopen( req )
  File "/usr/lib/python2.5/urllib2.py", line 124, in urlopen
    return _opener.open(url, data)
  File "/usr/lib/python2.5/urllib2.py", line 381, in open
    response = self._open(req, data)
  File "/usr/lib/python2.5/urllib2.py", line 399, in _open
    '_open', req)
  File "/usr/lib/python2.5/urllib2.py", line 360, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.5/urllib2.py", line 1107, in http_open
    return self.do_open(httplib.HTTPConnection, req)
  File "/usr/lib/python2.5/urllib2.py", line 1082, in do_open
    raise URLError(err)
urllib2.URLError: <urlopen error (110, 'Connection timed out')>
```

**Comment by javiplx** [comment]
When DNS resolution fails, the exception is neither captured:

```
root@BlusensFreePC10:~/Escritorio# aupd --random-wait 10 --force-upgrade pull
Traceback (most recent call last):
  File "/usr/bin/aupd", line 447, in <module>
    if not command[ subcmd ]( config , options , args ) :
  File "/usr/bin/aupd", line 323, in pull
    return amebaC3_client.client.pull ( _url , _uuid , _cmds.split(',') , outdated_retcode )
  File "/usr/lib/python2.6/dist-packages/amebaC3_client/client.py", line 125, in pull
    sessid , delay = login( url , uuid )
  File "/usr/lib/python2.6/dist-packages/amebaC3_client/client.py", line 64, in login
    res = urllib2.urlopen( req )
  File "/usr/lib/python2.6/urllib2.py", line 126, in urlopen
    return _opener.open(url, data, timeout)
  File "/usr/lib/python2.6/urllib2.py", line 391, in open
    response = self._open(req, data)
  File "/usr/lib/python2.6/urllib2.py", line 409, in _open
    '_open', req)
  File "/usr/lib/python2.6/urllib2.py", line 369, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.6/urllib2.py", line 1161, in http_open
    return self.do_open(httplib.HTTPConnection, req)
  File "/usr/lib/python2.6/urllib2.py", line 1136, in do_open
    raise URLError(err)
urllib2.URLError: <urlopen error [Errno -2] Name or service not known>
```

**Comment by javiplx** [comment]
Another "exotic" uncaptured exception

```
Traceback (most recent call last):
  File "/usr/bin/aupd", line 464, in <module>
    if not command[ subcmd ]( config , options , args ) :
  File "/usr/bin/aupd", line 337, in pull
    ret = amebaC3_client.client.pull ( _url , _uuid , _cmds.split(',') , outdated_retcode )
  File "/usr/lib/python2.5/site-packages/amebaC3_client/client.py", line 125, in pull
    sessid , delay = login( url , uuid )
  File "/usr/lib/python2.5/site-packages/amebaC3_client/client.py", line 64, in login
    res = urllib2.urlopen( req )
  File "/usr/lib/python2.5/urllib2.py", line 124, in urlopen
    return _opener.open(url, data)
  File "/usr/lib/python2.5/urllib2.py", line 381, in open
    response = self._open(req, data)
  File "/usr/lib/python2.5/urllib2.py", line 399, in _open
    '_open', req)
  File "/usr/lib/python2.5/urllib2.py", line 360, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.5/urllib2.py", line 1107, in http_open
    return self.do_open(httplib.HTTPConnection, req)
  File "/usr/lib/python2.5/urllib2.py", line 1080, in do_open
    r = h.getresponse()
  File "/usr/lib/python2.5/httplib.py", line 928, in getresponse
    response.begin()
  File "/usr/lib/python2.5/httplib.py", line 385, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python2.5/httplib.py", line 349, in _read_status
    raise BadStatusLine(line)
httplib.BadStatusLine
```

**Comment by javiplx** [comment]
We have seen this exception to happen when a connection is closed from the server side during processing.
To reproduce the error a long processing time must be forced at the server side

**Comment by javiplx** [comment]
Fixed on commit bbd5857d9a275ee404963505ba007cd98152eabe
